lib/morphix.ex

```diff
@@ -562,6 +562,9 @@ defmodule Morphix do
   iex> Morphix.compactiform!(%{nil_nil: nil, not_nil: "a value", nested: %{nil_val: nil, other: "other", nested_empty: %{}}})
   %{not_nil: "a value", nested: %{other: "other"}}
 
+  iex> Morphix.compactiform!([nil, "string", %{nil_nil: nil, not_nil: "a value", nested: %{nil_val: nil, other: "other", nested_empty: %{}}}, ["nested", nil, 2]])
```

Comment:
It seems like this should return `["string", %{not_nil: "a value", nested: %{other: "other"}}, ["nested", 2]]` instead of flattening the nested list.

Reply:
Thanks for catching this! The latest commit should fix this issue.